services/fuzzilli/fuzzilli.sh

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env bash
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+set -e
+set -x
+set -o pipefail
+
+function retry () {
+  op="$(mktemp)"
+  for _ in {1..9}; do
+    if "$@" >"$op"; then
+      cat "$op"
+      rm "$op"
+      return
+    fi
+    sleep 30
+  done
+  rm "$op"
+  "$@"
+}
+
+function tc-get-secret () {
+  TASKCLUSTER_ROOT_URL="${TASKCLUSTER_PROXY_URL-$TASKCLUSTER_ROOT_URL}" retry taskcluster api secrets get "project/fuzzing/$1"
+}
+
+# shellcheck source=recipes/linux/common.sh
+source $HOME/.local/bin/common.sh
+
+if [[ -z "$NO_SECRETS" ]]
+then
+  # setup AWS credentials to use S3
+  setup-aws-credentials
+fi
+
+if [[ -n "$TASK_ID" ]] || [[ -n "$RUN_ID" ]] ; then
+  function get-deadline () {
+    tmp="$(mktemp -d)"
+    retry taskcluster api queue task "$TASK_ID" >"$tmp/task.json"
+    retry taskcluster api queue status "$TASK_ID" >"$tmp/status.json"
+    deadline="$(date --date "$(jshon -e status -e deadline -u <"$tmp/status.json")" +%s)"
+    started="$(date --date "$(jshon -e status -e runs -e "$RUN_ID" -e started -u <"$tmp/status.json")" +%s)"
+    max_run_time="$(jshon -e payload -e maxRunTime -u <"$tmp/task.json")"
+    rm -rf "$tmp"
+    run_end="$((started + max_run_time))"
+    if [[ $run_end -lt $deadline ]]; then
+      echo "$run_end"
+    else
+      echo "$deadline"
+    fi
+  }
+  TARGET_TIME="$(($(get-deadline) - $(date +%s) - 5 * 60))"
+else
+  TARGET_TIME=$((10 * 365 * 24 * 3600))
+fi
+
+# Get the deploy key for fuzzilli from Taskcluster
+tc-get-secret deploy-fuzzilli | jshon -e secret -e key -u > $HOME/.ssh/id_rsa.fuzzilli
+tc-get-secret deploy-langfuzz-config | jshon -e secret -e key -u > $HOME/.ssh/id_rsa.langfuzz-config
+
+chmod 0600 $HOME/.ssh/id_rsa.*
+
+# Setup Key Identities
+cat << EOF > $HOME/.ssh/config
+
+Host fuzzilli
+Hostname github.com
+IdentityFile $HOME/.ssh/id_rsa.fuzzilli
+
+Host langfuzz-config
+Hostname github.com
+IdentityFile $HOME/.ssh/id_rsa.langfuzz-config
+EOF
+
+# -----------------------------------------------------------------------------
+
+cd $HOME
+
+retry git clone -v --depth 1 git@langfuzz-config:MozillaSecurity/langfuzz-config.git config
+retry git clone -v --depth 1 https://github.com/MozillaSecurity/FuzzManager/
+
+# Checkout the configuration with bootstrap script
+retry git clone -v --depth 1 git@fuzzilli:MozillaSecurity/fuzzilli.git fuzzilli
```

Comment:
`retry git clone ...` is problematic. If the initial clone fails, the subsequent retries will also fail because the destination already exists.

There is a `git-clone` function in `common.sh` which does a shallow clone with working retries.
```suggestion
git-clone git@langfuzz-config:MozillaSecurity/langfuzz-config.git config
git-clone https://github.com/MozillaSecurity/FuzzManager/

# Checkout the configuration with bootstrap script
git-clone git@fuzzilli:MozillaSecurity/fuzzilli.git fuzzilli
```